QueenSolutionConsole.cs

```diff
@@ -21,85 +21,78 @@ public class Program
         public static void Main(string[] args)
         {
             var sw = System.Diagnostics.Stopwatch.StartNew();
-
-            BuildBlankSet();
-            BuildStateLoops();
-
+  
             _Count = 0;
             _Spin = 0;
 
-            BuildNew(0, (_BoardSize + 1) / 2);
+            BuildNew(0, (_BoardSize + 1) / 2, 0, 0, 0);
+            
             sw.Stop();
             Console.WriteLine($"Board Size = {_BoardSize}  Total Spin={_Spin}");
             Console.WriteLine($"Time {sw.ElapsedMilliseconds} milliseconds  Solutions={_Solutions.Count}");
-           
-        }
 
-        //initialize static current solution array
-        private static void BuildBlankSet()
-        {
-            for (int j = 0; j < _BoardSize; j++)
-                _Current[j] = -1;
-        }
-        //initialize static stateful loop arrays
-        private static void BuildStateLoops()
-        {
-            for (int x = 0; x < _BoardSize; x++)
-            {
-                _Loops[x] = new int[_BoardSize];
-                for (int y = 0; y < _BoardSize; y++)
-                    _Loops[x][y] = y;
-            }
         }
 
+      
+   
         //recursive method to iterate all potential positions
-        //only include valid solutions
         //maintain state as we go so when we come back to a previous loop and iterate, we don't have to rebuild the entire state
         //only iterate through half as we use Y-flip to find other solutions
-        private static void BuildNew(int position, int end)
+        private static void BuildNew(int position, int end, long yline, long updiag, long downdiag)
         {
-            int[][] state = new int[_BoardSize][];
-            for (int x = 0; x < _BoardSize; x++)
-                state[x] = new int[_BoardSize];
+            long YLine = 0;
+            long UpDiag = 0;
+            long DownDiag = 0;
+            bool test = true;
+
+            long tempY = 0;
+            long tempUp = 0;
+            long tempDown = 0;
 
             for (int x = 0; x < end; x++)
             {
-                if (_Loops[position][x] >= 0)
-                {
-                    _Spin++;
-                    _Current[position] = x;
+                _Spin++;
+                _Current[position] = x;
 
+                tempY = 2 << x;
+                tempUp = 2 << (_BoardSize + x - position);
+                tempDown = 2 << (position + x);
+
+                test = (tempY & yline) != tempY;
+                test = test && ((tempDown & downdiag) != tempDown);
```

Comment:
Why not let C# compiler short-circuit?  Also, if you invert the test you can compare to 0 instead of comparing to temp variable.  Combining these might let you inline the test inside the `if ()`.

Reply:
another good observation.  I wrote it too quickly to think of testing against 0 (has to be more efficient I would imagine).